File: DemoMethods/src/main/java/net/ravendb/demo/basic/Delete.java
package net.ravendb.demo.basic;

import net.ravendb.client.IDocumentSession;
import net.ravendb.demo.DocumentStoreHolder;
import net.ravendb.demo.entities.Company;
import org.springframework.stereotype.Controller;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RequestParam;

@Controller
public class Delete {

    @RequestMapping("/Basic/Delete")
    public String delete(
            @RequestParam (value = "companyId", defaultValue = "2") int companyId) {
        try (IDocumentSession session = DocumentStoreHolder.getStore().openSession()) {
            session.delete(session.load(Company.class, companyId));
            session.saveChanges();

            return String.format("Company %s Deleted Successfully", companyId);
        }
    }
}

File: DemoMethods/PythonDemo/demo_methods/Basic/Delete.py
from demo.holder import DocumentStoreHolder
from demo.entities import *
from pyravendb.data.document_convention import DocumentConvention


class Delete(object):
    @staticmethod
    def delete(company_id=2):
        with DocumentStoreHolder.get_store().open_session() as session:
            company_id = "{0}/{1}".format(DocumentConvention.default_transform_type_tag_name(Company.__name__),
                                          company_id)
            session.delete(company_id)
            session.save_changes()
            return "CompanyId {0} Deleted  Successfully.".format(company_id)
